src/App.tsx

```diff
@@ -15,36 +27,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const sortFieldsArray: SortField[] = [
+  {
+    text: 'Sort alphabetically',
+    value: SortType.Alphabetically
+  },
+  {
+    text: 'Sort by length',
+    value: SortType.ByLength
+  }
+];
+
+function getSortGoods(goods: string[], sortField: SortType | null, isReversed: boolean): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case 'length':
+          return good1.length - good2.length;
+        case 'alphabetically':
+          return good1.localeCompare(good2);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | null>(null);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const visibleGoods = getSortGoods(goodsFromServer, sortField, isReversed);
+
+  const handleSort = (value: SortType | null) => {
+    if (value === 'reverse') {
+      setIsReversed(!isReversed);
+    } else {
+      setSortField(value);
+    }
+  };
+
+  const handleReset = () => {
+    setSortField(null);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
-
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
-
-        <button type="button" className="button is-warning is-light">
+        {sortFieldsArray.map(({ text, value }) => (
+          <button
+            type="button"
+            key={value}
+            className={classNames('button', {
+              'is-info': value === 'alphabetically',
+              'is-success': value === 'length',
+              'is-light': value !== sortField,
+            })}
+            onClick={() => handleSort(value)}
```

Comment:
```suggestion
            onClick={() => setSortField(value)}
```
just call it directly